cookiecutter/repository.py

```diff
@@ -41,6 +41,20 @@ def expand_abbreviations(template, abbreviations):
     return template
 
 
+def valid_repository(repo_directory):
```

Comment:
I'm not really happy with the result of this move:
- `valid_repository` now raises instead of just returning a `bool` and I was planning to use it again when [allowing already checked out cookiecutters to be referenced by name](https://github.com/audreyr/cookiecutter/issues/385#issuecomment-197502702).
- i don't like having to essentially process the json twice, it makes more sense for the validation to be where it's actually going to be used.
Ideas?